scripts/list-build-dependencies.sh

```diff
@@ -1,196 +1,55 @@
-				PACKAGES+=(lib32-sdl)
-			else
-				PACKAGES+=(sdl)
-			fi
-			;;
-
-		zypper)
-			# OpenSUSE offers 32-bit versions of SDL and SDL_net (but not others)
-			PACKAGES=(devel_basis libtool autoconf-archive opusfile)
-			if [[ "${BITS}" == 32 ]]; then
-				PACKAGES+=(libSDL-devel-32bit libSDL_net-devel-32bit)
-			else
-				PACKAGES+=(SDL SDL_net)
-			fi
-			;;
-
-		xcode)
-			# If the user doesn't want to use Homebrew or MacPorts, then they are limited to
-			# Apple's Clang plus the provided command line development tools provided by Xcode.
-			COMPILER=""
-			echo "Execute the following:"
-			echo "   xcode-select --install    # to install command line development tools"
-			echo "   sudo xcodebuild -license  # to accept Apple's license agreement"
-			echo ""
-			echo "Now download, build, and install the following manually to avoid using Homebrew or MacPorts:"
-			echo " - coreutils autogen autoconf automake pkg-config libpng sdl sdl_net opusfile"
-			;;
-
-		brew)
-			# If the user wants Clang, we knock it out because it's provided provided out-of-the-box
-			VERSION_DELIM="@"
-			if [[ "${COMPILER}" == "clang" ]]; then
-				COMPILER=""
-			fi
-			PACKAGES=(coreutils autogen autoconf autoconf-archive automake pkg-config libpng sdl sdl_net opusfile)
-			;;
-
-		macports)
-			PACKAGES=(coreutils autogen automake autoconf autoconf-archive pkgconfig libpng libsdl libsdl_net opusfile)
-			;;
-
-		msys2)
-			# MSYS2 only supports the current latest releases of Clang and GCC, so we disable version customization
-			COMPILER_VERSION=""
-			local pkg_type
-			pkg_type="x86_64"
-			if [[ "${BITS}" == 32 ]]; then
-				pkg_type="i686"
-			fi
-			PACKAGES=(autogen autoconf autoconf-archive base-devel automake-wrapper)
-			for pkg in pkg-config libtool libpng zlib SDL SDL_net opusfile; do
-				PACKAGES+=("mingw-w64-${pkg_type}-${pkg}")
-			done
-			COMPILER="mingw-w64-${pkg_type}-${COMPILER}"
-			;;
-
-		vcpkg)
-			# VCPKG doesn't provide Clang or GCC, so we knock out the compiler and just give packages
-			COMPILER=""
-			PACKAGES=(libpng sdl1 sdl1-net opusfile)
-			;;
-		*)
-			usage "Unknown package manager ${1}"
-			;;
-	esac
-
-	if [[ -n "${COMPILER_VERSION}" && -n "${COMPILER}" ]]; then
-		COMPILER+="${VERSION_DELIM}${COMPILER_VERSION}"
-	fi
-	echo "${COMPILER} ${PACKAGES[*]}"
-}
+	# Import our settings and report missing values
+	import manager "${manager:-}"
+	if [[ -z "${selected_type:-}" ]]; then arg_error "--compiler" "${TYPES[*]}"; fi
 
-function main() {
-	parse_args "$@"
-	list_packages "${PACKAGE_MANAGER}"
+	# If a version was provided then construct a postfix string given the manager's
+	# potential unique delimeter (ie: gcc@9 for brew, gcc-9 for apt).
+	# shellcheck disable=SC2034,SC2154
+	postfix=$([[ -n "${version:-}" ]] && echo "${delim}${version}" || echo "")
+	import "${selected_type:-}" "${manager:-}"
 }
+# shellcheck disable=SC2034
+data_dir="packages"
 
-main "$@"
+# shellcheck disable=SC1090,SC1091
```

Comment:
I tried but it didn't work; does it need absolute paths?  Also, in some cases rhe file imported is defined by a variable.

Reply:
I'll check - in the past I used to avoid these shellcheck warning via other method as well (placing files for sourcing in a PATH (I don't remember if PATH was modified dynamically or not though) - `source` will look for files in there if direct path is not given.

Reply:
```
# shellcheck source=scripts/automator/main.sh
source "$(dirname "$0")/automator/main.sh"
```
Works; I should have known that the path should be correct wrt the working directory; glad it's solved.  Might be able to fix a couple more now too.

Reply:
Was also able to fix it in build.sh, but couldn't do the automator 1090/1091 because it's fully dynamic.  I tried to feed it a known file, but it's too smart :-)

```
In scripts/automator/main.sh line 60:
if [[ -f "$varfile" ]]; then source "$varfile"
                                             ^-- SC1091: Not following: scripts/automator/packages/gcc-apt was not specified as input (see shellcheck -x).

```
And then disabling SC1091 gives:

```
In scripts/automator/main.sh line 60:
		if [[ -f "$varfile" ]]; then source "$varfile"
                                             ^-- SC1090: Can't follow non-constant source. Use a directive to specify location.
```

So it's disabled there.

Fixed in 89fbb403